### PR TITLE
fix: fix broken commands

### PR DESCRIPTION
**Pull request by kjin**
#49 broke `gts check` (and `gts fix`) by omitting a `module.exports` in `lint.ts`.

In addition to fixing the above, I've added tests for this, and run `npm run lint`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "format": "clang-format -i -style='{Language: JavaScript, BasedOnStyle: Google, ColumnLimit: 80}' src/*.ts test/*.ts",
     "lint": "tslint -c tslint.json  --project . --type-check -t codeFrame",
     "lint-fix": "tslint -c tslint.json --project .  --type-check -t codeFrame --fix",
+    "prepublish": "npm run compile",
```
Comment (ofrobots) [inline package.json]: I think `prepare` is perferred over `prepublish`, but that might be a seasonal thing (too lazy to find citation).

File: src/lint.ts [hunk 2]
```diff
@@ -48,3 +48,8 @@ export function createProgram(options: Options): ts.Program {
   const tsconfigPath = path.join(options.targetRootDir, 'tsconfig.json');
   return Linter.createProgram(tsconfigPath);
 }
+
+module.exports = {
```
Comment (ofrobots) [inline src/lint.ts]: Why is this needed? I think `lint` and `createProgram` are already exported, so the `require('./lint).lint` in `cli.ts` should work without this, unless I am mistaken.

File: package.json (hunk 1 above)
Comment (kjin) [reply package.json]: Yes, you're right -- changed

File: src/lint.ts (hunk 2 above)
Comment (kjin) [reply src/lint.ts]: Done -- seems like the root problem was simply omitting the `.lint` in the first place.

File: src/util.ts [hunk 3]
```diff
@@ -37,7 +37,7 @@ export function nop() {
 export async function getTSConfig(
     rootDir: string, parmReadFilep?: ReadFileP): Promise<any> {
   const tsconfigPath = path.join(rootDir, 'tsconfig.json');
-  const json = await (parmReadFilep || readFilep)(tsconfigPath, 'utf8');
+  const json = await(parmReadFilep || readFilep)(tsconfigPath, 'utf8');
```
Comment (ofrobots) [inline src/util.ts]: wat? did the formatter produce this? Seems wrong. Nothing you can address though.

Comment (kjin) [reply src/util.ts]: Yep -- likely a bug in the formatter. I've modified the code a little to prevent this.

File: src/util.ts [hunk 4]
```diff
@@ -35,9 +35,12 @@ export function nop() {
  * @param rootDir Directory where the tsconfig.json should be found.
  */
 export async function getTSConfig(
-    rootDir: string, parmReadFilep?: ReadFileP): Promise<any> {
+    rootDir: string, paramReadFilep?: ReadFileP): Promise<any> {
   const tsconfigPath = path.join(rootDir, 'tsconfig.json');
-  const json = await(parmReadFilep || readFilep)(tsconfigPath, 'utf8');
+  if (!paramReadFilep) {
```
Comment (ofrobots) [inline src/util.ts]: nit: more idiomatic to do `paramReadFilep = paramReadfilep || readFilep`. Can we rename the variable because it no longer contains just the parameter value though.

Comment (kjin) [reply src/util.ts]: I named it `customReadFilep`.
